File: Cpp/Source/Method.Test/source/AuthenticationServiceTest.cpp

#include "Method/include/AuthenticationService.h"

#include <gtest/gtest.h>

TEST( administratorIsAlwaysAuthenticated, AuthenticationServiceTest )
{
    AuthenticationService service;
    int adminId = 12345;
    EXPECT_TRUE(service.isAuthenticated(adminId));
}

TEST( normalUserIsNotAuthenticatedInitially, AuthenticationServiceTest )
{
    AuthenticationService service;
    int normalUserId = 11111;
    EXPECT_FALSE(service.isAuthenticated(normalUserId));
}

File: Java/src/test/java/method/AuthenticationServiceTest.java
package method;

import com.pduda.method.AuthenticationService;
import org.junit.Assert;
import org.junit.Test;

public class AuthenticationServiceTest {

    @Test
    public void administratorIsAlwaysAuthenticated() {
        AuthenticationService service = new AuthenticationService();
        int adminId = 12345;
        Assert.assertTrue(service.isAuthenticated(adminId));
    }
    @Test
    public void normalUserIsNotAuthenticatedInitially() {
        AuthenticationService service = new AuthenticationService();
        int normalUserId = 11111;
        Assert.assertFalse(service.isAuthenticated(normalUserId));
    }

}
